homeassistant/components/yandex_transport/yamaps.py

```diff
@@ -0,0 +1,133 @@
+"""Library to fetch info about stops from Yandex Maps."""
+import json
+import re
+from urllib.parse import urlencode
+
+import aiohttp
+
+from homeassistant.exceptions import InvalidStateError
+from homeassistant.helpers.aiohttp_client import SERVER_SOFTWARE as HA_USER_AGENT
+
+RESOURCE = "https://yandex.ru/maps/api/masstransit/getStopInfo"
+AJAX_KEY = "ajax"
+LANG_KEY = "lang"
+LOCALE_KEY = "locale"
+MODE_KEY = "mode"
+SIGNATURE = "s"
+
+CONFIG = {
+    "init_url": "https://maps.yandex.ru",
+    "uri": RESOURCE,
+    "params": {AJAX_KEY: 1, LANG_KEY: "ru", LOCALE_KEY: "ru_RU", MODE_KEY: "prognosis"},
+    "headers": {"User-Agent": HA_USER_AGENT},
+}
+PARAMS = "params"
+ID_KEY = "id"
+URI_KEY = "uri"
+CSRF_TOKEN_KEY = "csrfToken"
+SESSION_KEY = "sessionId"
+
+SCHEMA = [
+    AJAX_KEY,
+    CSRF_TOKEN_KEY,
+    ID_KEY,
+    LANG_KEY,
+    LOCALE_KEY,
+    MODE_KEY,
+    SESSION_KEY,
+    URI_KEY,
+]
+
+
+class YandexMapsRequester:
```

Comment:
The device specific interface code and data parsing should be part of a standalone library published on PyPI.

https://developers.home-assistant.io/docs/creating_component_code_review#4-communication-with-devicesservices

Reply:
I extracted this code to the library on PyPI